### PR TITLE
d3d12shader.rs

### PR DESCRIPTION
**Pull request by red75prime**
ID3D11ShaderReflection::GetRequiresFlags() returns UINT64, so I changed
 type of D3D_SHADER_REQUIRES_... constants to UINT64 in d3d11shader.rs

### CODE REVIEW

File: src/d3d12shader.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    D3D12_SHVER_VERTEX_SHADER = 0x1,
+    D3D12_SHVER_GEOMETRY_SHADER = 0x2,
+    D3D12_SHVER_HULL_SHADER = 0x3,
+    D3D12_SHVER_DOMAIN_SHADER = 0x4,
+    D3D12_SHVER_COMPUTE_SHADER = 0x5,
+    D3D12_SHVER_RESERVED0 = 0xFFF0,
+}}
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct D3D12_FUNCTION_DESC {
```
Comment (retep998) [inline src/d3d12shader.rs]: Please use `STRUCT!`

Comment (red75prime) [inline src/d3d12shader.rs]: Done

**Comment by red75prime** [comment]
appveyor is over capacity, so the build failed

**Comment by retep998** [comment]
This branch is out of date and needs a rebase.

File: src/d3d12shader.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
+// Copyright © 2016; Dmitry Roschin
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
```
Comment (retep998) [inline src/d3d12shader.rs]: Winapi is currently only licensed under MIT. Could you perhaps update this to match what the rest of the source files do?

Comment (red75prime) [inline src/d3d12shader.rs]: Done

**Pull request merged**